### PR TITLE
JSON & alpha

### PR DESCRIPTION
**Pull request by LuukSantegoeds**
# Description

We created our own JSON library from an external library. This wouldn't by adding the submodule to CMake so we chose this workaround. For this library we created an adapter which fits all of our needs (for now at least). This adapter can throw exceptions when used wrong and it contains tests.

Furthermore some modifications has been done to the rendering to add alpha (transparency) to platforms.

There is currently an issue with the Texture Manager and Alpha. When you change the alpha of one texture, because it is only loaded once, every object will have that alpha. Please comment what you think.
 
## Screenshot(s)
_No_

# How can this be tested?

This PR can be tested when combined with [this](https://github.com/Brick-Studios/BeastArena/pull/27) PR from BeastArena. Also the tests can be run. If you want, you could tinker around with the JSON files in the engine or the test file.

- [x] The engine compiles without errors or warnings
- [x] All tests succeed

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Included 1 or more screenshots
- [x] Code is const correct

### CODE REVIEW

**Comment by Peter-JanGootzen** [comment]
As for the alpha mode question, you should put an alpha value in the base Renderable and use SDL_GetRenderDrawBlendMode and SDL_SetRenderDrawBlendMode to apply and deapply the alpha value in the [DrawScreen()](https://github.com/Brick-Studios/BrickEngine/blob/master/lib/rendering/renderer.cpp#L58).

File: include/brickengine/json/exceptions/noPathException.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include <iostream>
+#include <exception>
+
+class NoPathException : public std::exception {
+private:
+  std::string m_msg;
+public:
+  NoPathException() : m_msg(std::string("String needs to be a valid path and confirmed via the boolean")){}
+
+  virtual const char* what() const throw() {
```
[the remark this answers is not in the document]
Comment (Bram-Boris) [reply include/brickengine/json/exceptions/noPathException.hpp]: Erm, see this: https://bb.avans.nl/bbcswebdav/pid-1533592-dt-content-rid-52048127_1/courses/AII-1920A-INCPPLS1/CPPLS1.6(1).pdf

[the remark this answers is not in the document]
Comment (JanSchollaert) [reply include/brickengine/json/exceptions/noPathException.hpp]: ahja

File: lib/rendering/renderables/renderable.cpp [hunk 2]
```diff
@@ -2,8 +2,11 @@
 #include "brickengine/rendering/renderables/data/rect.hpp"
 #include "SDL2/SDL.h"
 
-Renderable::Renderable(int layer) :
-    layer(layer) {}
+Renderable::Renderable(int layer) : layer(layer) {
+    alpha = 255;
```
Comment (Peter-JanGootzen) [inline lib/rendering/renderables/renderable.cpp]: You can use
`: layer(layer), alpha(255) {}`

Comment (Bram-Boris) [reply lib/rendering/renderables/renderable.cpp]: Ah cool, did not think of this.

**Pull request merged**